pyscf/cc/ccsd.py

```diff
@@ -1042,6 +1046,7 @@ def init_amps(self, eris=None):
         time0 = logger.process_clock(), logger.perf_counter()
         if eris is None:
             eris = self.ao2mo(self.mo_coeff)
+            self.e_hf = self.get_e_hf(mo_coeff=self.mo_coeff)
```

Comment:
if eris is specified in input, self.e_hf may not be properly created.

Reply:
if self.e_hf is None, e_hf will be evaluated new, otherwise e_hf = self.e_hf. e_hf is needed in this function.